### PR TITLE
Add checks for UTF-8

### LINKED ISSUE #78: babi should always open files using UTF-8
especially on windows, this can result in mojibake

ensure that the `open` calls go through `encoding='UTF-8'` -- maybe also validate stdin does the same

### PR DESCRIPTION
**Pull request by 7brokenmirrors**
Resolves #78

Add "encoding='UTF-8'" to open() calls
Add check if stdin can be decoded as UTF-8

I wasn't sure how reading from stdin would be utilized so I might totally miss the point here

### CODE REVIEW

**Comment by 7brokenmirrors** [comment]
Oh I guess I have to add a test so we can get back to 100 %

File: babi/main.py [hunk 1]
```diff
@@ -133,7 +134,11 @@ def main(argv: Optional[Sequence[str]] = None) -> int:
 
     if '-' in args.filenames:
         print('reading stdin...', file=sys.stderr)
-        stdin = sys.stdin.read()
+        stdin_bytes = sys.stdin.buffer.read()
+        try:
+            stdin = decode(stdin_bytes, encoding='UTF-8')
+        except UnicodeError:
+            raise
```
Comment (asottile) [inline babi/main.py:141]: you can just leave out the `try except` it'll raise by itself

File: babi/history.py [hunk 2]
```diff
@@ -19,7 +19,10 @@ def save(self) -> Generator[None, None, None]:
         history_dir = xdg_data('history')
         os.makedirs(history_dir, exist_ok=True)
         for filename in os.listdir(history_dir):
-            with open(os.path.join(history_dir, filename)) as f:
+            with open(
+                os.path.join(history_dir, filename),
+                encoding='UTF-8',
+            ) as f:
```
Comment (asottile) [inline babi/history.py:25]: can save a line and the odd indentation by doing:

```python
history_filename = os.path.join(history_dir, filename)
with open(history_filename, encoding='UTF-8') as f:
```

(same below)

File: babi/main.py [hunk 3]
```diff
@@ -133,7 +134,11 @@ def main(argv: Optional[Sequence[str]] = None) -> int:
 
     if '-' in args.filenames:
         print('reading stdin...', file=sys.stderr)
-        stdin = sys.stdin.read()
+        stdin_bytes = sys.stdin.buffer.read()
```
Comment (asottile) [inline babi/main.py:137]: you can write this as `sys.stdin.buffer.read().decode()` -- the default encoding in python3 is UTF-8

**Comment by 7brokenmirrors** [comment]
Thanks for the feedback, I implemented your suggestions.

I really wanted to write a test for this but couldn't get stdin to work, sadly.

**Pull request merged**